packages/amplify-cli-core/src/utils/packageManager.ts

```diff
@@ -9,41 +9,103 @@ import { BuildType } from '@aws-amplify/amplify-function-plugin-interface';
 /**
  * package managers type
  */
-export type PackageManagerType = 'yarn' | 'npm' | 'pnpm';
+export type PackageManagerType = 'yarn' | 'npm' | 'pnpm' | 'custom';
 const packageJson = 'package.json';
 
 /**
  * package Manager type
  */
-export type PackageManager = {
+export interface PackageManager {
   packageManager: PackageManagerType;
   lockFile: string;
   executable: string;
+  displayValue: string;
   version?: SemVer;
-  displayValue?: string;
-};
+  getRunScriptArgs: (scriptName: string) => string[];
+  getInstallArgs: (buildType: BuildType) => string[];
+}
 
 const isWindows = process.platform === 'win32';
 
+export class NpmPackageManager implements PackageManager {
```

Comment:
should these classes be internal to this package so they have to be retrieved using `getPackageManagerByType`?

Reply:
will do